map-low.go

```diff
@@ -99,29 +123,70 @@ func GetMapByID(id uint32) (*BPFMapLow, error) {
 	}, nil
 }
 
-// GetMapsIDsByName searches for maps with a given name.
-// It returns a slice of unsigned 32-bit integers representing the IDs of matching maps.
-// If no maps are found, it returns an empty slice and no error.
-func GetMapsIDsByName(name string) ([]uint32, error) {
-	bpfMapsIds := []uint32{}
+// GetMapNextID retrieves the next available map ID after the given startID.
+// It returns the next map ID and an error if one occurs during the operation.
+func GetMapNextID(id uint32) (uint32, error) {
+	startID := C.uint(id)
+	retC := C.bpf_map_get_next_id(startID, &startID)
+	if retC == 0 {
+		return uint32(startID), nil
+	}
 
-	startId := C.uint(0)
-	nextId := C.uint(0)
+	errno := syscall.Errno(-retC)
+	switch errno {
+	case syscall.ENOENT:
+		return uint32(startID), GetMapNextIDENOENTError{}
+	case syscall.EINVAL:
+		return uint32(startID), GetMapNextIDEINVALError{}
+	case syscall.EPERM:
+		return uint32(startID), GetMapNextIDEPERMError{}
+	default:
+		return uint32(startID), fmt.Errorf("failed to get next map id: %w", errno)
+	}
```

Comment:
I realized that creating our own error types would increase maintenance efforts and ignore the original error message. So how about just returning the original encapsulated error? We know they are syscall.EXX, so the caller can compare them directly. WDYT?

```suggestion
	return uint32(startID), fmt.Errorf("failed to get next map id: %w", errno)
```